libs/@local/harpc/net/src/session/server/session_id.rs

```diff
@@ -0,0 +1,30 @@
+pub(crate) struct SessionIdProducer {
+    next: usize,
+}
+
+impl SessionIdProducer {
+    #[must_use]
+    pub(crate) const fn new() -> Self {
+        Self { next: 0 }
+    }
+
+    pub(crate) fn produce(&mut self) -> SessionId {
+        let id = self.next;
+        self.next = self.next.wrapping_add(1);
+
+        SessionId(id)
+    }
+}
+
+// what is in the session is not part of the service, but higher level!
```

Comment:
What do you mean?

Reply:
Whoops that isn't supposed to be there, that was me brainstorming. I first wanted put the session logic into the transport layer, then decided against it. This was a question to future me that I forgot to remove :P